### ISSUE TITLE
VSync has no effect on Linux and macOS

### ISSUE THREAD

**Issue by alexey-lysiuk** [nukeykt/NBlood#30]
Regardless of VSync setting's value it doesn't seem to do anything on Linux (tested on Kubuntu 18.04) and macOS. It works correctly on Windows though.

`SDL_GL_SetSwapInterval()` returns zero and `SDL_GL_SetSwapInterval()` return one. These are the expected results while FPS counter shows quite the opposite with `r_maxfps` set to zero.

Unrelated to VSync, I must admit that the FPS limiter itself is an odd thing. It fully consumes one CPU core instead of waiting on an appropriate synchronization primitive. But it's a totally different issue.

**Comment by alexey-lysiuk** [comment]
After rechecking Linux and Mac builds again it appeared that I misinterpreted the issue.

With `r_maxfps 0` and `r_swapinterval 1` Linux build behaves like it should. Apparently, I did something wrong during the first try.

Mac build is another story. If it was compiled with SDK older than 10.14 or if it runs on any macOS older than Mojave, everything works as expected.
Although, if it runs on macOS 10.14 and it was built with SDK of the same version (i.e. by Xcode 10.x), glitches manifest themselves. I noticed excess CPU load, blurry picture, and non-functional VSync.

They have nothing to do with NBlood/EDuke32 or even SDL. These bugs are tightly coupled with recent Apple's "inventions" like OpenGL deprecation, dark mode, and forcing of Core Animation layer backing.
Unfortunately, I'm not aware of a good solution for those problems.
